helix-term/src/commands.rs

```diff
@@ -2082,6 +2082,28 @@ mod cmd {
         Ok(())
     }
 
+    fn setting(
+        cx: &mut compositor::Context,
+        args: &[&str],
+        _event: PromptEvent,
+    ) -> anyhow::Result<()> {
+        use toml::value::Map;
+        use toml::Value;
+
+        let mut current_conf: Map<String, Value> =
+            toml::de::from_slice(&toml::ser::to_vec(&cx.editor.config)?)?;
+        let conf = toml::from_str::<Map<String, Value>>(&args.join(" "))?;
+        for (key, value) in conf {
+            if let Some(current_value) = current_conf.get_mut(&key) {
+                *current_value = value;
+            } else {
+                bail!("Key `{}` does not exists", key)
+            }
+        }
+        cx.editor.config = toml::de::from_slice(&toml::ser::to_vec(&current_conf)?)?;
```

Comment:
I think you can do it this way, create a static keys where it is only generated once and reuse it. It will be used for keys completion.

Once the key is specified, get the value and serialize only that to be displayed, after reading the value from the user, serialize it to be saved and directly set the specific config.

What do you think? That way we can minimized what we serialize/deserialize, but I still think it is needed for it to be displayed in the info box.

Reply:
> I think you can do it this way, create a static keys where it is only generated once and reuse it. It will be used for keys completion.

Yes, this is smart! Thank you :smile: 

> after reading the value from the user, serialize it to be saved and directly set the specific config.

But how can I set only the specific config?
Should I make a big `match` on the key, or do you have another idea in mind?

Reply:
> But how can I set only the specified config?

Since you already have the keys, you can just match it I believe?